### ISSUE TITLE
Check for required system packages (e.g. tkinter) at the start of the build process

### ISSUE THREAD

**Issue by jimkring** [Nuitka/Nuitka#2750]
As a user, knowing exactly which prerequisite packages are going to be needed for my build, up front, would be extremely helpful and would save me a lot of time when setting up a new build process (especially when working with containers).

For example, here's where I currently experienced some "pain":

I have a Nuikta project build that is configured with `--plugin-enable=tk-inter`.  I'm trying to build my project on the `quay.io/pypa/manylinux2014_x86_64` container image (for better glibc compatibility), which doesn't have a lot of packages pre-installed -- it doesn't have `tkinter` installed.

About 10 minutes into my project's nuikta build process, I get the following error:

```
573.5 FATAL: tk-inter: Could not find Tcl, you might need to use '--tcl-library-dir' and if that works, report a bug so it can be added to Nuitka.
576.4 Nuitka-Reports: Compilation crash report written to file 'nuitka-crash-report.xml'.
```

So, it turns out I need to `yum install -y tkinter tcl` and possibly some other system package(s).

After doing all this, I wonder...  _could Nuitka (and/or the `tk-inter` plugin for Nuitka) checked for this before starting the build and then gave me the error message before it spent 10 minutes on other compilation tasks?_

As a user, knowing exactly which prerequisite packages are going to be needed for my build, up front, would be extremely helpful.  It would be great if Nuikta (and it's plugins) could provide me with information about this (and maybe even offer to install the required packages), as early as possible after I start typing `python -m nuikta...` :)

**Comment by kayhayen** [comment]
There is like 3 things involved here.

The tk-inter plugin, like all plugins of Nuitka tries to be smart and be allowed to be enabled until it is proven that it's not going to be used. That is to allow for all plugins to be enabled by default.

When it as a GUI plugin is enabled, it expects to be present, as it will have suppressed Qt usage, and so on. As a GUI plugin it is not enabled by default therefore, but assumes that its activation is a special deed and then is not tolerant about anything anymore.

For this plugin, we could during the init phase, make it check DLL and data files presence. What i would really want there to be is for this plugin to be yaml config, but there we generally are very tolerant of absence of DLLs and data files, so that's not going to help at all. We cannot be insisting on DLLs and data files generally, as there is not a bunch of things to do.

What I wonder, is how it would even happen. But these guys trigger it:

```
def _isTkInterModule(module):
    full_name = module.getFullName()
    return full_name in ("Tkinter", "tkinter", "PySimpleGUI", "PySimpleGUI27")
```

I am assuming, that a tkinter import would have failed, and you are using one of the later, i.e. PySimpleGUI? The code ought to just not run unless that is the case. I assume we can make it crash if none of it exists.

Also, why did this not fire for you?

```
        self.tk_inter_version = getTkInterVersion()

        if self.tk_inter_version is None:
            self.sysexit("Error, it seems tk-inter is not installed.")
```

with it being like this:

```
def getTkInterVersion():
    """Get the tk-inter version or None if not installed."""
    try:
        if str is bytes:
            return str(__import__("TkInter").TkVersion)
        else:
            return str(__import__("tkinter").TkVersion)
    except ImportError:
        # This should lead to no action taken ideally.
        return None
```

So it would have to be installed?

Given the crash report, I am very confused here, that yum install shouldn't have changed anything.

**Comment by kayhayen** [comment]
You have either observed a bug in the code, or are mistaken? Without the crash report it's hard to tell what it is. Nuitka may also just not have your Python supported there for Tcl. I e.g. only recently added Homebrew paths, and there is always some that it may not yet do automatically. The goal of course being to do them all automatically and never error out with the error you encountered.

**Comment by kayhayen** [comment]
Any feedback on this one Jim, I would love to just have things work in all configurations, it seems you had one which is not handled, and we can add it @jimkring

**Comment by jimkring** [comment]
@kayhayen sorry for the delay in response and than you for your thoughtful reponse/ideas. I'd like to eventually figure out how to reproduce this and give you more info for a fix. I'm not sure when I'll have the time to reproduce it, since I worked through my issue and it does take a long time to test (10 min per iteration), so maybe we can leave this issue open and put it in the "ice box" for later when I'm back working on that project where I experienced it. If anyone else runs into a similar issue, in the meantime, LMK and I can assist :)
